File: GymCRM.BillingAPI/Models/DTOs/Payment.cs
namespace GymCRM.BillingAPI.Models.DTOs;

public class Payment
{
    public Guid Id { get; set; }
    public Guid SubscriptionId { get; set; }
    public decimal Amount { get; set; }
    public int Method { get; set; }
    public int Status { get; set; }
    public DateTime PaidAt { get; set; }
    public string? ExternalReference { get; set; }
    public DateTime DateCreated { get; set; }
}

File: gymcrm.web/src/modules/billing/types/payment.ts
// Mirrors GymCRM.BillingAPI.Models.DTOs.Payment 1:1.
export interface Payment {
    id: string;
    subscriptionId: string;
    amount: number;
    method: number;
    status: number;
    paidAt: string;
    externalReference?: string;
    dateCreated: string;
}
